### PR TITLE
update: add rule tags to metrics

### PR DESCRIPTION
**Pull request by jasondellaluce**
Signed-off-by: Jason Dellaluce <jasondellaluce@gmail.com>

**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area pkg

**What this PR does / why we need it**:
After the recent merge of https://github.com/falcosecurity/falco/pull/1714, Falco gRPC `output` service now has a field containing rule tags. This PR follows up to those updates by including rule tags in the exported metrics too, which has been an open issue for a while (see https://github.com/falcosecurity/falco-exporter/issues/53). 

**Which issue(s) this PR fixes**:
Fixes https://github.com/falcosecurity/falco-exporter/issues/53

**Additional notes**:
Rule tags are implemented as a label inside the `falco_events` metric. The label is a simple concatenation of all the tags, separated by a comma (`,`). This has the benefit of not increasing the metric cardinality, which is a big concern in Prometheus. The tradeoff is that querying by rule tags requires the usage of a regex.

A little optimization has been adopted by adding commas at the beginning and at the end of the concatenated string. In this way, the `tags` label can be queried with a simpler regex like `.*,tag,.*` instead of `(^|.*,)tag(,.*|$)`. This pattern is well explained here: https://www.robustperception.io/little-things-matter.

An example of tag-based query is reported below:
```
falco_events{tags=~".*,filesystem,.*"}
```

### CODE REVIEW

**Comment by jasondellaluce** [comment]
This is still a WIP until the next release of https://github.com/falcosecurity/client-go, which will presumably happen after Falco's 0.30.0 release. Module dependencies will need to be updated to include the changes introduced in https://github.com/falcosecurity/client-go/pull/55.

**Comment by jasondellaluce** [comment]
This PR has been unblocked after the release of [client-go@v0.4.0](https://pkg.go.dev/github.com/falcosecurity/client-go@v0.4.0/pkg/client), so it is now ready for a review.

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,8 @@ module github.com/falcosecurity/falco-exporter
 go 1.14
 
 require (
-	github.com/falcosecurity/client-go v0.3.0
-	github.com/prometheus/client_golang v1.9.0
+	github.com/falcosecurity/client-go v0.4.0
```
Comment (leogr) [inline go.mod:6]: @jasondellaluce @leodido Thank you for that! :hugs:

**Review by leogr — APPROVED** [review]
/approve

**Pull request merged**